### PR TITLE
Updated Electrokinetics Docu to python, removed Ek_Reaction from docu

### LINKED ISSUE #1249: electrokinetics.rst: Convert to Py interface

### PR DESCRIPTION
**Pull request by KunTares**
Fixes #1249

Description of changes:
 - Updated Electrokinetics Docu to python, removed Ek_Reaction from docu

### CODE REVIEW

File: doc/sphinx/advanced_methods.rst [hunk 1]
```diff
@@ -1105,6 +1103,318 @@ with respect to the underlying object, the triangle 123 needs to be
 properly oriented (as explained in the section on volume in
 oif_global_forces interaction).
 
+.. _Electrokinetics:
+
+Electrokinetics
+---------------
+
+The electrokinetics setup in allows for the description of
```
Comment (KaiSzuttor) [inline doc/sphinx/advanced_methods.rst]: |es| missing

File: doc/sphinx/advanced_methods.rst [hunk 2]
```diff
@@ -1105,6 +1103,318 @@ with respect to the underlying object, the triangle 123 needs to be
...
+electro-hydrodynamic systems on the level of ion density distributions
+coupled to a Lattice-Boltzmann (LB) fluid. The ion density distributions
+may also interact with explicit charged particles, which are
+interpolated on the LB grid. In the following paragraph we briefly
+explain the electrokinetic model implemented in , before we come to the
```
Comment (KaiSzuttor) [inline doc/sphinx/advanced_methods.rst]: |es| missing

File: doc/sphinx/advanced_methods.rst [hunk 3]
```diff
@@ -1105,6 +1103,318 @@ with respect to the underlying object, the triangle 123 needs to be
...
+description of the interface.
+
+If you are interested in using the electrokinetic implementation in for
```
Comment (KaiSzuttor) [inline doc/sphinx/advanced_methods.rst]: |es| missing

File: doc/sphinx/advanced_methods.rst [hunk 4]
```diff
@@ -1105,6 +1103,318 @@ with respect to the underlying object, the triangle 123 needs to be
...
+scientific purposes, please contact G. Rempfer before you start your
```
Comment (KaiSzuttor) [inline doc/sphinx/advanced_methods.rst]: outdated

File: doc/sphinx/advanced_methods.rst [hunk 5]
```diff
@@ -1105,6 +1103,318 @@ with respect to the underlying object, the triangle 123 needs to be
...
+:math:`\eta`
+    the dynamic viscosity of the fluid,
+
+:math:`\eta_{\text{b}}`
+    the bulk viscosity of the fluid.
+
+The temperature :math:`T`, and diffusion constants :math:`D_k` and
+mobilities :math:`\nu_k` of individual species are linked through the
+Einstein-Smoluchowski relation :math:`D_k /
+\nu_k = {k_\mathrm{B}T}`. The system of equations described in Eqs. -,
```
Comment (KaiSzuttor) [inline doc/sphinx/advanced_methods.rst]: Eq reference missing

File: doc/sphinx/advanced_methods.rst [hunk 6]
```diff
@@ -1105,6 +1103,318 @@ with respect to the underlying object, the triangle 123 needs to be
...
+~~~~~
+
+.. _Initialization:
+
+Initialization
+^^^^^^^^^^^^^^
+::
+
+    import espressomd
+    sys = espressomd.System()
```
Comment (KaiSzuttor) [inline doc/sphinx/advanced_methods.rst]: please change to new syntax

File: doc/sphinx/advanced_methods.rst [hunk 7]
```diff
@@ -1105,6 +1103,318 @@ with respect to the underlying object, the triangle 123 needs to be
...
+    ek = espressomd.electrokinetics.Electrokinetics(agrid = 1.0, lb_density = 1.0, 
+    viscosity = 1.0, friction = 1.0, T =1.0, prefactor = 1.0, stencil = 'linkcentered', advection = True, fluid_coupling = 'friction')
+    sys.actors.add(ek)
+
+.. note:: `Feature ELECTROKINETICS and LB_GPU required`
+
+The above is a minimal example how to initialize the LB fluid, and
+it is very similar to the Lattice-Boltzmann command in set-up. We
+therefore refer the reader to Chapter :ref:`Lattice-Boltzmann` for details on the
+implementation of LB in and describe only the major differences here.
```
Comment (KaiSzuttor) [inline doc/sphinx/advanced_methods.rst]: |es| missing

File: doc/sphinx/advanced_methods.rst [hunk 8]
```diff
@@ -1105,6 +1103,318 @@ with respect to the underlying object, the triangle 123 needs to be
...
+
+The first major difference with the LB implementation is that the
+electrokinetics set-up is a Graphics Processing Unit (GPU) only
+implementation. There is no Central Processing Unit (CPU) version, and
+at this time there are no plans to make a CPU version available in the
+future. To use the electrokinetics features it is therefore imperative
+that your computer contains a CUDA capable GPU which is sufficiently
+modern.
+
+To set up a proper LB fluid using this command one has to specify at
```
Comment (KaiSzuttor) [inline doc/sphinx/advanced_methods.rst:1268]: |es| missing

File: doc/sphinx/advanced_methods.rst [hunk 9]
```diff
@@ -1105,6 +1103,318 @@ with respect to the underlying object, the triangle 123 needs to be
...
+least the following options: ``agrid``, ``lb_density``, ``viscosity``, ``friction``, ``T``, and ``prefactor``. The other options can be
+used to modify the behavior of the LB fluid. Note that the command does
+not allow the user to set the time step parameter as is the case for the
+Lattice-Boltzmann command, this parameter is instead taken directly from the value set for
+``espressomd.System().time_step``. The LB `mass density` is set independently from the
```
Comment (KaiSzuttor) [inline doc/sphinx/advanced_methods.rst]: remove parantheses for System (its not a method) and use ```:attr:`espressomd.System.time_step` ```

File: doc/sphinx/advanced_methods.rst [hunk 10]
```diff
@@ -1105,6 +1103,318 @@ with respect to the underlying object, the triangle 123 needs to be
...
+
+
+The feature `EK_ELECTROSTATIC_COUPLING` enables the action of the electrostatic potential due to the
+electrokinetics species and charged boundaries on the MD particles. The
+forces on the particles are calculated by interpolation from the
+electric field which is in turn calculated from the potential via finite
+differences. This only includes interactions between the species and
+boundaries and MD particles, not between MD particles and MD particles.
+To get complete electrostatic interactions a particles Coulomb method
+like Ewald or P3M has to be activate too.
```
Comment (KaiSzuttor) [inline doc/sphinx/advanced_methods.rst]: typo: activated

File: doc/sphinx/advanced_methods.rst [hunk 11]
```diff
@@ -1105,6 +1103,318 @@ with respect to the underlying object, the triangle 123 needs to be
...
+.. note:: `Feature EK_BOUNDARIES required`
+
+The EKBoundary command allows one to set up (internal or external) boundaries for
+the electrokinetics algorithm in much the same way as the command is
+used for the LB fluid. The major difference with the LB command is given
+by the option ``charge_density``, with which a boundary can be endowed with a volume
+charge density. To create a surface charge density, a combination of two
+oppositely charged boundaries, one inside the other, can be used.
+However, care should be taken to maintain the surface charge density
+when the value of is changed. Currently, the following shapes are available:
```
Comment (KaiSzuttor) [inline doc/sphinx/advanced_methods.rst]: value of what?

File: doc/sphinx/advanced_methods.rst [hunk 12]
```diff
@@ -1105,6 +1103,318 @@ with respect to the underlying object, the triangle 123 needs to be
...
+wall, sphere, cylinder, rhomboid, pore, stomatocyte, hollow\_cone, and
```
Comment (KaiSzuttor) [inline doc/sphinx/advanced_methods.rst]: pore is not available anymore, simple pore?

File: doc/sphinx/advanced_methods.rst [hunk 13]
```diff
@@ -1105,6 +1103,318 @@ with respect to the underlying object, the triangle 123 needs to be
...
+spherocylinder. We refer to the documentation of the lbboundary command
+(Chapter :ref:`Using shapes as Lattice-Boltzmann boundary`) for information on the options associated to these shapes. In order to properly set up the boundaries, the ``charge_density`` and ``shape``
```
Comment (KaiSzuttor) [inline doc/sphinx/advanced_methods.rst]: is the link up to date?

**Review by KaiSzuttor — CHANGES_REQUESTED** [review]

File: doc/sphinx/advanced_methods.rst (hunk 6 above)
Comment (KunTares) [reply doc/sphinx/advanced_methods.rst]: What is the new syntax?

File: doc/sphinx/advanced_methods.rst (hunk 8 above)
Comment (KunTares) [reply doc/sphinx/advanced_methods.rst:1268]: Where is the |es| missing?

File: doc/sphinx/advanced_methods.rst (hunk 6 above)
Comment (KaiSzuttor) [reply doc/sphinx/advanced_methods.rst]: box_l as constructor parameter

File: doc/sphinx/advanced_methods.rst (hunk 8 above)
Comment (KaiSzuttor) [reply doc/sphinx/advanced_methods.rst:1268]: i misread the sentence, nothing is missing

**Review by fweik — APPROVED** [review]

**Pull request merged**